tests/models/test_taskinstance.py

```diff
@@ -1743,6 +1743,32 @@ def on_execute_callable(context):
         ti.refresh_from_db()
         assert ti.state == State.SUCCESS
 
+    def test_success_callback_handles_and_logs_exception(self):
+        called = completed = False
+
+        def on_success_callable(context):
+            nonlocal called, completed
+            called = True
+            raise KeyError
+            completed = True
+
+        dag = DAG(
+            'test_success_callback_handles_exception',
+            start_date=DEFAULT_DATE,
+            end_date=DEFAULT_DATE + datetime.timedelta(days=10),
+        )
+        task = DummyOperator(
+            task_id='op', email='test@test.test', on_success_callback=on_success_callable, dag=dag
+        )
```

Comment:
```suggestion
    def test_success_callback_handles_and_logs_exception(self, dag_maker):
        called = completed = False

        def on_success_callable(context):
            nonlocal called, completed
            called = True
            raise KeyError
            completed = True

        with dag_maker(
            'test_success_callback_handles_exception',
            end_date=DEFAULT_DATE + datetime.timedelta(days=10),
        ) :
              task = DummyOperator(
                  task_id='op', email='test@test.test', on_success_callback=on_success_callable
              )
        dag_maker.create_dagrun()
```
we recently introduced the `dag_maker` fixture so that we can reduce code duplication and also create dagrun whenever we create a dag.

Reply:
Oh, super neat! However, it looks like TestTaskInstance is a subclass of `unitest.Testcase`, which [doesn't seem to support all uses of pytest fixtures](https://docs.pytest.org/en/6.2.x/unittest.html#pytest-features-in-unittest-testcase-subclasses).  Is there a reason why some test classes are subclasses of unittest.testcase, and others aren't? 

Anyways, I've refactored this test to include all three callback types, but have been having some trouble getting the `dag_maker` fixture to work. Are you able to give me a hand with this?

Reply:
Oh, never mind. Use the previous method you used. I will need to work on this in a separate PR so we stop using unittest.

Reply:
Ok, is the plan to stop using `unittest` everywhere? I don't mind refactoring the tests in this file, but I'd rather do it in a separate PR.

Reply:
Yes. We want to change to using pytest and dag_maker. I'm already working on this file. You can check another one. Thanks @SamWheating